### PR TITLE
[feat]: [CDS-54154]: Jsch version upgrade

### PR DESCRIPTION
**Pull request by arvind-choudhary-h**
## Describe your changes
Here, we are upgrading the Jsch library to `mwiede/jsch` which will solve supporting new hashing algorithms. Included in this PR:
1. Library upgrade via Workspace file and related bazel build files
2. Defaulting to new Ssh client implementation if key type is vault ssh (as older implementation will not support vault with new library)
3. Supporting older SshSessionFactory to work with newer library incase we have to switch FF.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50664)

### CODE REVIEW

**Comment by arvind-choudhary-h** [comment]
trigger uts

**Review by lovish1999 — DISMISSED** [review]
LGTM

File: 960-api-services/src/main/java/io/harness/shell/SshSessionFactory.java [hunk 1]
```diff
@@ -175,6 +175,8 @@ public static Session fetchSSHSession(SshSessionConfig config, LogCallback logCa
       }
     }
     session.setConfig("StrictHostKeyChecking", "no");
+    session.setConfig("server_host_key", session.getConfig("server_host_key") + ",ssh-rsa");
```
[the remark this answers is not in the document]
Comment (arvind-choudhary-h) [reply 960-api-services/src/main/java/io/harness/shell/SshSessionFactory.java:181]: Thanks @acasian-harness for pointing this out. I missed this in the JschClient file. 
Anyways, so this is required for backward compatibility where we enable SSH/RSA1 signatures if server supports these keys. By default in the new lib, this is disabled.
Ref: https://github.com/mwiede/jsch/blob/master/Readme.md#why-do-ssh-rsa-type-keys-not-work-with-this-jsch-fork-and-my-server

**Review by ivan-83 — DISMISSED** [review]
:lgtm:

> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50664#-:-N_jRPZMB5-TPh6teyga:bnfp4nl)* status: 1. unresolved discussions. Waiting on team(s): del-team.

**Review by acasian-harness — DISMISSED** [review]

**Review by ivan-83 — APPROVED** [review]
> Reviewed 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50664)* status: 1. unresolved discussions. Waiting on team(s): del-team.

**Review by acasian-harness — APPROVED** [review]

**Comment by arvind-choudhary-h** [comment]
trigger review

**Comment by arvind-choudhary-h** [comment]
trigger messagecheck

**Comment by arvind-choudhary-h** [comment]
trigger ut5,ut6

**Comment by arvind-choudhary-h** [comment]
trigger messagecheck

**Review by jennyjames21 — APPROVED** [review]

**Review by jamie-harness — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
332-ci-manager change LG

**Review by gn-harness — APPROVED** [review]

**Review by pranay-harness — DISMISSED** [review]
Approved as part of https://harness.atlassian.net/browse/SECAPP-1043

**Comment by arvind-choudhary-h** [comment]
This is all reviewed but here comes conflict for the second time. As many teams are involved in the review and its Friday, I'll ask manager to FM this.

Screenshot 2023-07-28 at 1 02 52 PM

**Comment by arvind-choudhary-h** [comment]
trigger uts

**Comment by arvind-choudhary-h** [comment]
trigger ut7

**Comment by arvind-choudhary-h** [comment]
The failing test is fixed in https://github.com/harness/harness-core/pull/50962/files

**Pull request merged**